src/tests/tracing/eventpipe/diagnosticport/diagnosticport.cs

```diff
@@ -403,10 +403,9 @@ public static async Task<bool> TEST_CanGetProcessInfo2WhileSuspended()
                     }
                     else if (TestLibrary.Utilities.IsNativeAot)
                     {
-                        // shipping criteria: no EVENTPIPE-NATIVEAOT-TODO left in the codebase
-                        // https://github.com/dotnet/runtime/issues/83051
-                        // NativeAOT currently always returns empty string
-                        Utils.Assert(processInfo2.ManagedEntrypointAssemblyName == string.Empty);
+                        string expectedName = System.Reflection.Assembly.GetExecutingAssembly().GetName().Name;
```

Comment:
I think this could become the `else` branch. I'd delete the `if (TestLibrary.Utilities.IsNativeAot)` two lines above and delete the `else` block starting at line 410. The `else` block is single file unfriendly for no good reason.